examples/hikari_lightbulb_basic_queue/music_plugin.py

```diff
@@ -0,0 +1,255 @@
+
+PREFIX = "!" # Should probably set this to an environment variable
+TOKEN = os.environ["DISCORD_TOKEN"]
+LAVALINK_PASSWORD = os.environ["LAVALINK_PASSWORD"]
+
+# if True connect to voice with the hikari gateway instead of lavasnek_rs's
+HIKARI_VOICE = False
+
+
+class EventHandler:
+    """Events from the Lavalink server"""
+
+    async def track_start(self, _lava_client, event):
+        logging.info("Track started on guild: %s", event.guild_id)
+
+    async def track_finish(self, _lava_client, event):
+        logging.info("Track finished on guild: %s", event.guild_id)
+
+    async def track_exception(self, lavalink, event):
+        logging.warning("Track exception event happened on guild: %d", event.guild_id)
+
+        # If a track was unable to be played, skip it
+        skip = await lavalink.skip(event.guild_id)
+        node = await lavalink.get_guild_node(event.guild_id)
+
+        if not skip:
+            await event.message.respond("Nothing to skip")
+        else:
+            if not node.queue and not node.now_playing:
+                await lavalink.stop(event.guild_id)
+
+
+class Music(lightbulb.Plugin):
+    def __init__(self, bot):
+        super().__init__()
+        self.bot = bot
+
+    @lightbulb.listener(hikari.ShardReadyEvent)
+    async def start_lavalink(self, event) -> None:
+        """Event that triggers when the hikari gateway is ready."""
+
+        builder = (
+            lavasnek_rs.LavalinkBuilder(self.bot.get_me().id, TOKEN)
+            .set_host('127.0.0.1')
+            .set_password(LAVALINK_PASSWORD)
+        )
+
+        if HIKARI_VOICE:
+            builder.set_start_gateway(False)
+
+        lava_client = await builder.build(EventHandler())
+
+        self.bot.data.lavalink = lava_client
+
+
+    @lightbulb.check(lightbulb.guild_only)
+    @lightbulb.command()
+    async def join(self, ctx) -> int:
+        """Join the user's voice channel and create a lavalink session."""
+
+        states = self.bot.cache.get_voice_states_view_for_guild(ctx.get_guild())
+        voice_state = list(
+            filter(lambda i: i.user_id == ctx.author.id, states.iterator())
+        )
+        
+        if not voice_state:
+            await ctx.respond("Connect to a voice channel first")
+            return 0
+
+        channel_id = voice_state[0].channel_id
+
+        if HIKARI_VOICE:
+            await self.bot.update_voice_state(ctx.guild_id, channel_id, self_deaf=True)
+            connection_info = await self.bot.data.lavalink.wait_for_full_connection_info_insert(
+                ctx.guild_id
+            )
+        else:
+            try:
+                connection_info = await self.bot.data.lavalink.join(ctx.guild_id, channel_id)
+            except TimeoutError:
+                await ctx.respond("I was unable to connect to the voice channel, maybe missing permissions? or some internal issue.")
+                return 0
+
+        await self.bot.data.lavalink.create_session(connection_info)
```

Comment:
On the pure hikari example, all of this is on a separate method, so the play() command doesn't send the notification about it having joined a channel.

Reply:
Changed to reflect the pure hikari example

Reply:
You didn't do this.